ProjectSQ.Tests/ProcessorServiceTests/CallReturnTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Xunit;
+using FluentAssertions;
+using ProjectSQ.Services;
+using ProjectSQ.Interfaces.Memory;
+using Moq;
+using Microsoft.AspNetCore.SignalR;
+using ProjectSQ.Models;
+
+namespace ProjectSQ.Tests.ProcessorServiceTests
+{
+    public class CallReturnTests
+    {
+        private readonly ProcessorService _processorService;
+
+        public CallReturnTests()
+        {
+            Mock<IHubContext<RealTimeHub>> mockHubContext = new();
+            Mock<IMemoryService> mockMemoryService = new();
+            _processorService = new ProcessorService(mockHubContext.Object, mockMemoryService.Object);
+            _processorService.ResetData();
+        }
+
+        [Fact]
+        public void Call_ValidFunctionName_SetsCurrentInstruction()
+        {
+            // Arrange
+            Memory.internalMemory = new string[]
+            {
+                "main",
+                "mov reg1, 2",
+                "call func1",
+                "function func1",
+                "add reg1, reg2"
```

Comment:
next time use the method parseService.LoadInstructions("ProjectSQ.Utils.input.txt");